src/editor/Store/ConfigurationStoreConnector.re

```diff
@@ -31,7 +31,20 @@ let start = () => {
       switch (configPath) {
       | Ok(configPathAsString) =>
         switch (ConfigurationParser.ofFile(configPathAsString)) {
-        | Ok(v) => dispatch(Actions.ConfigurationSet(v))
+        | Ok(v) =>
+          dispatch(Actions.ConfigurationSet(v));
+
+          switch (cliOptions) {
+          | Some(cliOptions) =>
+            let zenModeSingleFile =
+              Configuration.getValue(c => c.zenModeSingleFile, v);
+
+            if (zenModeSingleFile && List.length(cliOptions.filesToOpen) == 1) {
```

Comment:
This will run on every configuration reload, so also when you have a lot of files open. The toggle should only be done on startup.

Reply:
Well, yeah, that's what I mentioned in the description :smile: 

The question is how? Should I add a ref(boolean) to the closure and flip it?

Reply:
why not do this in the `initConfigurationEffect`?

Reply:
I would need to read and parse the config file there as well. So then we parse it twice unless I thread it through to `reloadConfigurationEffect` by parameterizing `Actions.ConfigurationReload` with and `option(Configuration.t)` :thinking:

Reply:
I moved it, currently we just read and parse the config file twice, on the other hand. Maybe I don't need to ... nevermind found a solution. If I dispatch `Actions.ConfigurationSet(configuration)` in `initConfigurationEffect` instead of `Actions.ConfigurationReload` we don't load the config twice.

Reply:
Let me know what you think.

Reply:
Nice catch @tcoopman , and thanks for investigating a fix @despairblue. Latest iteration looks good - it seems like it only uses the `zenModeSingleFile` on first load (in combination with the cli options passed in). Looks good to me! 👍